### PR TITLE
botocore 1.27.59

### PR DESCRIPTION
**Pull request by SC426**
s3fs 2020.10 --> aiobotocore 2.4.0 --> botocore 1.27.59 #53

Jira ticket: https://anaconda.atlassian.net/browse/PKG-154
Upstream repo: https://github.com/boto/botocore
Changelog: https://github.com/boto/botocore/tree/1.27.59/.changes

### CODE REVIEW

**Comment by skupr-anaconda** [comment]
> Linter check failed, with the following output: The following problems have been found:
> 
> ERROR: /tmp/abs_a8vrwt8eyh/clone/recipe/meta.yaml:10: invalid_url: https://pypi.org/packages/source/b/botocore/botocore-1.27.59.tar.gz : URL domain redirect pypi.org -> files.pythonhosted.org Errors were found

@SC426 You can ignore this lint check

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
-{% set version = "1.27.28" %}
-{% set hash = "b00da6290d8d42c6c3048b045d6537421e8b83e03b4f24bb44c5ea9f37ab6258" %}
+{% set version = "1.27.59" %}
+{% set hash = "eda4aed6ee719a745d1288eaf1beb12f6f6448ad1fa12f159405db14ba9c92cf" %}
 
 package:
   name: botocore
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: You can update source/url as `https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/botocore-{{ version }}.tar.gz`

Comment (skupr-anaconda) [inline recipe/meta.yaml]: Can be updated:
```
{% set name = "botocore" %}
{% set version = "1.27.59" %}
{% set hash = "eda4aed6ee719a745d1288eaf1beb12f6f6448ad1fa12f159405db14ba9c92cf" %}

package:
  name: {{ name|lower }}
  version: {{ version }}
```
see a `grayskull` web-variant: https://www.marcelotrevisani.com/grayskull

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
-{% set version = "1.27.28" %}
-{% set hash = "b00da6290d8d42c6c3048b045d6537421e8b83e03b4f24bb44c5ea9f37ab6258" %}
+{% set version = "1.27.59" %}
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:2]: Could we update to the latest **v1.28.2**? The requirements are the same https://github.com/boto/botocore/blob/1.28.2/setup.cfg

**Comment by skupr-anaconda** [comment]
After `botocore` you need to build `boto3` too:
`botocore ---> boto3`
see https://github.com/anaconda-distribution/distro-handbook/blob/main/source/package_review.rst

**Review by varlackc — COMMENTED** [review]
`JsonSchema` and `Coverage` seem to be mentioned as development requirements, the question might be if this dependencies are optional or not. 
https://github.com/boto/botocore/blob/1.27.59/requirements-dev.txt#L1&L4

There is also a dependency called `behave` that we do not seem to have used historically. 
Just curious on how do we want to tackle this in the future. 
https://github.com/boto/botocore/blob/1.27.59/requirements-dev.txt#L2

**Comment by marcoesters** [comment]
> After `botocore` you need to build `boto3` too: `botocore ---> boto3` see https://github.com/anaconda-distribution/distro-handbook/blob/main/source/package_review.rst

This version of botocore appears to be compatible with our current boto3: https://github.com/boto/boto3/blob/1.24.28/setup.py#L16

**Comment by marcoesters** [comment]
> `JsonSchema` and `Coverage` seem to be mentioned as development requirements, the question might be if this dependencies are optional or not. https://github.com/boto/botocore/blob/1.27.59/requirements-dev.txt#L1&L4
> 
> There is also a dependency called `behave` that we do not seem to have used historically. Just curious on how do we want to tackle this in the future. https://github.com/boto/botocore/blob/1.27.59/requirements-dev.txt#L2

I don't see a reason to include them. They are for upstream development and are not required to build, run, or test the package. For example, we have no interest in creating coverage reports.

File: recipe/meta.yaml (hunk 1 above)
Comment (SC426) [reply recipe/meta.yaml:5]: The build is not successful with this link

Comment (SC426) [reply recipe/meta.yaml]: This has been updated

Comment (SC426) [reply recipe/meta.yaml]: This has been updated

File: recipe/meta.yaml (hunk 2 above)
Comment (SC426) [reply recipe/meta.yaml:2]: `aiobotocore` needs this version.

https://github.com/aio-libs/aiobotocore/blob/fbfeb1242ab1f378f6d4fe84b5d3dacdd8874189/setup.py#L10

**Review by skupr-anaconda — APPROVED** [review]
Do not forget to remove before merging: 
```
channels:
  scrass_org: s3fs
```

**Review by cbouss — APPROVED** [review]
LGTM

**Pull request merged**